### PR TITLE
UI: Ember 2.18 -> Ember 3.0

### PR DESCRIPTION
**Pull request by DingoEatingFuzz**
This is step one in a process to upgrade to the latest Ember.js LTS version (3.4).

Major ember upgrades aren't as scary as minor upgrades, so this is a relatively safe PR.

The bulk of these changes were done with [codemods](https://github.com/ember-cli/ember-cli-update-codemods-manifest/blob/v3/manifest.json) that rewrote tests to the new testing style. The codemods aren't perfect and couldn't know about some of the custom testing stuff in Nomad, so there is also ~1000 lines of changes made by hand.

### CODE REVIEW

File: ui/config/environment.js [hunk 1]
```diff
@@ -21,7 +21,7 @@ module.exports = function(environment) {
     },
 
     APP: {
-      blockingQueries: true,
+      blockingQueries: false,
```
Comment (alisdair) [inline ui/config/environment.js]: Is this change related to the upgrade?

File: ui/tests/unit/services/breadcrumbs-test.js [hunk 2]
```diff
@@ -14,15 +14,23 @@ const makeRoute = (crumbs, controller = {}) =>
...
   });
 
-moduleFor('service:breadcrumbs', 'Unit | Service | Breadcrumbs', {
-  beforeEach() {
+module('Unit | Service | Breadcrumbs', function(hooks) {
+  setupTest(hooks);
+
+  hooks.beforeEach(function() {
+    this.subject = function() {
+      return this.owner.factoryFor('service:breadcrumbs').create();
```
Comment (alisdair) [inline ui/tests/unit/services/breadcrumbs-test.js]: This boilerplate for unit testing a service is kind of a bummer, but understandable.

File: ui/tests/helpers/codemirror.js [hunk 3]
```diff
@@ -18,9 +16,16 @@ export function getCodeMirrorInstance(container) {
...
-export default function registerCodeMirrorHelpers() {
-  registerHelper('getCodeMirrorInstance', function(app, selector) {
-    const helper = getCodeMirrorInstance(app.__container__);
-    return helper(selector);
+export default function setupCodeMirror(hooks) {
+  hooks.beforeEach(function() {
+    this.getCodeMirrorInstance = getCodeMirrorInstance(this.owner);
+
+    // Expose to window for access from page objects
+    window.getCodeMirrorInstance = this.getCodeMirrorInstance;
```
Comment (meirish) [inline ui/tests/helpers/codemirror.js]: 👌 nice

File: ui/tests/unit/services/breadcrumbs-test.js (hunk 2 above)
Comment (DingoEatingFuzz) [reply ui/tests/unit/services/breadcrumbs-test.js]: There may be a better way to do this, such as something like `setupSerializerTest(hooks)`, but I don't want to explore that until I make it to v3.4.

File: ui/tests/helpers/codemirror.js (hunk 3 above)
Comment (DingoEatingFuzz) [reply ui/tests/helpers/codemirror.js]: Lol, it's not my proudest code, but Mirage also does this in a less obvious place. That's all the excuse I needed 😄

File: ui/config/environment.js (hunk 1 above)
Comment (DingoEatingFuzz) [reply ui/config/environment.js]: Nope! Good catch.
